javascript/cw_tool.referrer.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * @file
+ *  Add referrer related behaviours.
+ */
+var CW = CW || {};
+
+(function($) {
+
+  'use strict';
+
+  CW.Referrer = $.extend({}, CW.Path, {
```

Comment:
This contains a knowledge about loading order and dependency. Either we should finally integrate Require.js (or equivalent) or we should add a loader func to the php module file so it's loaded the right order and with dependency.